django_socio_grpc/servicer_proxy.py

```diff
@@ -14,117 +21,175 @@
+    message: Message
+    context: GRPCSocioProxyContext
+    action: str
+    proxy: "ServicerProxy"
+
+
 class ServicerProxy:
-    def __init__(self, ServiceClass, **initkwargs):
-        self.service_class = ServiceClass
+    def __init__(self, service_class: Type["Service"], **initkwargs):
+        self.service_class = service_class
         self.initkwargs = initkwargs
 
-    def call_handler(self, action):
+        self.load_middleware(is_async=grpc_settings.GRPC_ASYNC)
+
+    def load_middleware(self, is_async=False):
+        """
+        Populate middleware lists from settings.MIDDLEWARE.
+        Must be called after the environment is fixed (see __call__ in subclasses).
+        From django BaseHandler.load_middleware
+        """
+        self._exception_middleware = []
+
+        get_response = self._get_response_async if is_async else self._get_response
+        handler = convert_exception_to_response(get_response)
+        handler_is_async = is_async
+        for middleware_path in reversed(grpc_settings.GRPC_MIDDLEWARE):
+            middleware = import_string(middleware_path)
+            middleware_can_sync = getattr(middleware, "sync_capable", True)
+            middleware_can_async = getattr(middleware, "async_capable", False)
+            if not middleware_can_sync and not middleware_can_async:
+                raise RuntimeError(
+                    "Middleware %s must have at least one of "
+                    "sync_capable/async_capable set to True." % middleware_path
+                )
+            elif not handler_is_async and middleware_can_sync:
+                middleware_is_async = False
+            else:
+                middleware_is_async = middleware_can_async
+            try:
+                # Adapt handler, if needed.
+                adapted_handler = self.adapt_method_mode(
+                    middleware_is_async,
+                    handler,
+                    handler_is_async,
+                    debug=settings.DEBUG,
+                    name="middleware %s" % middleware_path,
+                )
+                mw_instance = middleware(adapted_handler)
+            except MiddlewareNotUsed as exc:
+                if settings.DEBUG:
+                    if str(exc):
+                        logger.debug("MiddlewareNotUsed(%r): %s", middleware_path, exc)
+                    else:
+                        logger.debug("MiddlewareNotUsed: %r", middleware_path)
+                continue
+            else:
+                handler = adapted_handler
+
+            if mw_instance is None:
+                raise ImproperlyConfigured(
+                    "Middleware factory %s returned None." % middleware_path
+                )
+
+            handler = convert_exception_to_response(mw_instance)
+            handler_is_async = middleware_is_async
+
+        # Adapt the top of the stack, if needed.
+        handler = self.adapt_method_mode(is_async, handler, handler_is_async)
```

Comment:
this is already done above. Why doing it again ? Can you add a comment for that ?